### PR TITLE
fix: return reference to temporary variable and other compile warnings

### PR DESCRIPTION
**Pull request by gerard-queralt**

### CODE REVIEW

File: Source/DataModels/Components/ComponentMeshRenderer.cpp [hunk 1]
```diff
@@ -685,28 +685,28 @@ const bool ComponentMeshRenderer::IsTransparent() const
 	return material->IsTransparent();
 }
 
-const std::shared_ptr<ResourceTexture>& ComponentMeshRenderer::GetDiffuse() const
+std::shared_ptr<ResourceTexture> ComponentMeshRenderer::GetDiffuse() const
 {
 	return material->GetDiffuse();
 	;
```
Comment (PaablooCH) [inline Source/DataModels/Components/ComponentMeshRenderer.cpp:691]: there is a wild semicon here

**Review by Piterga12 — APPROVED** [review]

File: Source/FileSystem/Importers/ParticleSystemImporter.cpp [hunk 2]
```diff
@@ -71,13 +71,13 @@ void ParticleSystemImporter::Save
...
-		const ParticleEmitter* emitter = resource->GetEmitter(i);
-		size += emitter->GetName().size();
+		const ParticleEmitter* emitter = resource->GetEmitter(static_cast<unsigned int>(i));
+		size += static_cast<unsigned int>(emitter->GetName().size());
 		if(emitter->GetParticleTexture() != nullptr)
 		{
 			size += sizeof(UID);
 		}
-		size += (sizeof(int) + sizeof(bool)) * emitter->GetModules().size();
+		size += (sizeof(int) + sizeof(bool)) * static_cast<unsigned int>(emitter->GetName().size());
```
Comment (markmeliaa) [inline Source/FileSystem/Importers/ParticleSystemImporter.cpp:80]: This got changed from `GetModules` to `GetName`, is that expected?

Comment (gerard-queralt) [reply Source/FileSystem/Importers/ParticleSystemImporter.cpp:80]: Nope, guess I didn't pay enough attention copy pasting. Will fix asap, thanks for noticing 🙂

**Comment by markmeliaa** [comment]
Forgot about this, sorry! There seems to be a conflict now @gerard-queralt

**Review by markmeliaa — APPROVED** [review]

**Review by UriMurcia — APPROVED** [review]
Everything seems good to me, nice work!

**Pull request merged**